rllib/evaluation/env_runner_v2.py

```diff
@@ -791,6 +792,8 @@ def _handle_done_episode(
                     env_id
                 ],
                 env_index=env_id,
+                # Create new episode under this env_id.
+                next_episode=self._active_episodes[env_id],
```

Comment:
ok, I chatted with wes. if this is what they need, we don't really need to introduce a new callback, since doing ``self._active_episodes[env_id]`` here would trigger the ``on_episode_start()`` call.
essentially, what we are doing here is moving the ``on_episode_start()`` call to happen before reset().
so all we have to do is to make the ``self._active_episodes[env_id]`` call here, with a comment that says ``Assign the policy mapping for the next episode before env.reset() call``.
does this make sense?

Reply:
As per our discussion with Wes yesterday.
* We add a new callback: `on_episode_created`, which is triggered right after the Episode(V2)? instance has been instantiated, but before(!) the sub-environment is reset.
* Then we reset the sub-environment.
* Then we trigger `on_episode_start` (like we did before). This way the `on_episode_start` behavior remains unaltered/no API changes.